images/build/Dockerfiles/gitHubActions.Dockerfile

```diff
@@ -98,16 +98,29 @@ RUN ${IMAGES_DIR}/retry.sh "curl -o /usr/local/share/ca-certificates/verisign.cr
     
 # Install PHP pre-reqs	# Install PHP pre-reqs
 RUN if [ "${DEBIAN_FLAVOR}" = "buster" ]; then \
-    apt-get update \
-    && apt-get upgrade -y \
-    && apt-get install -y \
-        $PHPIZE_DEPS \
-        ca-certificates \
-        curl \
-        xz-utils \
-        libsodium-dev \
-        libncurses5 \
-    --no-install-recommends && rm -r /var/lib/apt/lists/* ; \
+        apt-get update \
+        && apt-get upgrade -y \
+        && apt-get install -y \
+            $PHPIZE_DEPS \
+            ca-certificates \
+            curl \
+            xz-utils \
+            libsodium-dev \
+            libncurses5 \
+        --no-install-recommends && rm -r /var/lib/apt/lists/* \
+        # A workaround for php 8.0 composer issue: https://github.com/microsoft/Oryx/issues/1100
+        && ln -s /usr/lib/x86_64-linux-gnu/libonig.so /usr/lib/x86_64-linux-gnu/libonig.so.4 \
```

Comment:
It seems that this line creates `libonig.so.4`, which is linked from the php binary, as a symbolic link to `libonig.so` forcefully.
But maybe `libonig.so` is a symbolic link to `libonig.so.5` in Kudu/buster.
I think this causes the following warnings.

```
php: Symbol `OnigEncodingEUC_JP' has different size in shared object, consider re-linking
php: Symbol `OnigEncodingUTF8' has different size in shared object, consider re-linking
```

Is this really a safe way?

Is it hard to re-build the php binary, installed by Oryx's [snippet](https://github.com/microsoft/Oryx/blob/e805ef4a86e4edb65a4bd0bf4f5a851a8c26c562/src/BuildScriptGenerator/PlatformInstallerBase.cs#L46) from `https://oryx-cdn.microsoft.io/php/php-8.0.x.tar.gz` , for buster?

Reply:
Hi @horihiro , thanks for the info. Yes I saw the warnings, I'm trying different kinds of workarounds and this one looks most promising. Messing around with the PHP binary is not advisable, the current binary applies with the same way as what we previously built these binaries. Please let us know if there's other advice or solutions.

Reply:
I found the [php binary](https://oryx-cdn.microsoft.io/php/php-buster-8.0.17.tar.gz) (php-**buster**-8.0.xyz.tar.gz) which was built for buster at [this](https://github.com/microsoft/Oryx/blob/main/platforms/php/buildPhp.sh#L43).

Is it hard to change a php binary which is downloaded at the following line to the above one?
https://github.com/microsoft/Oryx/blob/e805ef4a86e4edb65a4bd0bf4f5a851a8c26c562/src/BuildScriptGenerator/PlatformInstallerBase.cs#L57

Reply:
We build the php binary from the script and then publish it to the CDN storage account you mentioned. The binary script consists of lots of dependencies, changing or modifying components in here may causes some other issues, which is hard to predict. 
https://github.com/microsoft/Oryx/blob/main/platforms/php/prereqs/build.sh

Reply:
If switching `.so` link with PHP for stretch on buster is more predictable than using PHP for buster, it's OK.

Reply:
At least, PHP8/Laravel8 app can be deployed from VSCode with Oryx build and work without any problem after replacing binaries in `/tmp/oryx/platforms/php/8.0.17` and `/tmp/oryx/platforms/php-composer/2.0.8` to ones for buster on Oryx CDN manually.

<details>
<summary> Deployment logs </summary>

```
7:01:13 AM ***************: Starting deployment...
7:01:14 AM ***************: Creating zip package...
7:01:15 AM ***************: Zip package size: 273 kB
7:01:20 AM ***************: Fetching changes.
7:01:21 AM ***************: Cleaning up temp folders from previous zip deployments and extracting pushed zip file /tmp/zipdeploy/b962d9a9-2a52-4b31-a472-6772413cb888.zip (0.26 MB) to /tmp/zipdeploy/extracted
7:01:34 AM ***************: Updating submodules.
7:01:35 AM ***************: Preparing deployment for commit id '9517f275-2'.
7:01:36 AM ***************: Repository path is /tmp/zipdeploy/extracted
7:01:36 AM ***************: Running oryx build...
7:01:36 AM ***************: Command: oryx build /tmp/zipdeploy/extracted -o /home/site/wwwroot --platform php --platform-version 8.0 -i /tmp/8da24abab593c37 --log-file /tmp/build-debug.log 
7:01:37 AM ***************: Operation performed by Microsoft Oryx, https://github.com/Microsoft/Oryx
7:01:37 AM ***************: You can report issues at https://github.com/Microsoft/Oryx/issues
7:01:37 AM ***************: Oryx Version: 0.2.20211207.1, Commit: 46633df49cc8fbe9718772a3c894df221273b2af, ReleaseTagName: 20211207.1
7:01:37 AM ***************: Build Operation ID: |ZTzLalNcC/A=.1928273a_
7:01:37 AM ***************: Repository Commit : 9517f275-209c-4aeb-9247-6f0d03ed29ea
7:01:37 AM ***************: Detecting platforms...
7:01:40 AM ***************: Detected following platforms:
7:01:40 AM ***************:   nodejs: 14.19.1
7:01:40 AM ***************:   php: 8.0.17
7:01:40 AM ***************: Using intermediate directory '/tmp/8da24abab593c37'.
7:01:40 AM ***************: Copying files to the intermediate directory...
7:01:40 AM ***************: Done in 0 sec(s).
7:01:40 AM ***************: Source directory     : /tmp/8da24abab593c37
7:01:40 AM ***************: Destination directory: /home/site/wwwroot
7:01:41 AM ***************: PHP executable: /tmp/oryx/platforms/php/8.0.17/bin/php
7:01:41 AM ***************: Composer archive: /tmp/oryx/platforms/php-composer/2.0.8/composer.phar
7:01:41 AM ***************: Running 'composer install --ignore-platform-reqs --no-interaction'...
7:01:42 AM ***************: Installing dependencies from lock file (including require-dev)
7:01:42 AM ***************: Verifying lock file contents can be installed on current platform.
7:01:42 AM ***************: Package operations: 110 installs, 0 updates, 0 removals
7:01:42 AM ***************:   - Downloading doctrine/inflector (2.0.4)
7:01:42 AM ***************:   - Downloading doctrine/lexer (1.2.3)
7:01:42 AM ***************:   - Downloading symfony/polyfill-ctype (v1.25.0)
7:01:42 AM ***************:   - Downloading webmozart/assert (1.10.0)
7:01:42 AM ***************:   - Downloading dragonmantank/cron-expression (v3.3.1)
7:01:42 AM ***************:   - Downloading symfony/polyfill-php80 (v1.25.0)
7:01:42 AM ***************:   - Downloading symfony/polyfill-mbstring (v1.25.0)
7:01:42 AM ***************:   - Downloading symfony/var-dumper (v5.4.6)
7:01:42 AM ***************:   - Downloading symfony/polyfill-intl-normalizer (v1.25.0)
7:01:42 AM ***************:   - Downloading symfony/polyfill-intl-grapheme (v1.25.0)
7:01:42 AM ***************:   - Downloading symfony/string (v6.0.3)
7:01:42 AM ***************:   - Downloading symfony/deprecation-contracts (v3.0.1)
7:01:42 AM ***************:   - Downloading psr/container (1.1.2)
7:01:42 AM ***************:   - Downloading symfony/service-contracts (v2.5.1)
7:01:42 AM ***************:   - Downloading symfony/polyfill-php73 (v1.25.0)
7:01:42 AM ***************:   - Downloading symfony/console (v5.4.7)
7:01:42 AM ***************:   - Downloading psr/log (2.0.0)
7:01:42 AM ***************:   - Downloading monolog/monolog (2.5.0)
7:01:42 AM ***************:   - Downloading voku/portable-ascii (1.6.1)
7:01:42 AM ***************:   - Downloading phpoption/phpoption (1.8.1)
7:01:42 AM ***************:   - Downloading graham-campbell/result-type (v1.0.4)
7:01:42 AM ***************:   - Downloading vlucas/phpdotenv (v5.4.1)
7:01:42 AM ***************:   - Downloading symfony/css-selector (v6.0.3)
7:01:42 AM ***************:   - Downloading tijsverkoyen/css-to-inline-styles (2.2.4)
7:01:43 AM ***************:   - Downloading symfony/routing (v5.4.3)
7:01:43 AM ***************:   - Downloading symfony/process (v5.4.7)
7:01:43 AM ***************:   - Downloading symfony/polyfill-php72 (v1.25.0)
7:01:43 AM ***************:   - Downloading symfony/polyfill-intl-idn (v1.25.0)
7:01:43 AM ***************:   - Downloading symfony/mime (v5.4.7)
7:01:43 AM ***************:   - Downloading symfony/http-foundation (v5.4.6)
7:01:43 AM ***************:   - Downloading psr/event-dispatcher (1.0.0)
7:01:43 AM ***************:   - Downloading symfony/event-dispatcher-contracts (v3.0.1)
7:01:43 AM ***************:   - Downloading symfony/event-dispatcher (v6.0.3)
7:01:43 AM ***************:   - Downloading symfony/error-handler (v5.4.7)
7:01:43 AM ***************:   - Downloading symfony/http-kernel (v5.4.7)
7:01:43 AM ***************:   - Downloading symfony/finder (v5.4.3)
7:01:43 AM ***************:   - Downloading symfony/polyfill-iconv (v1.25.0)
7:01:43 AM ***************:   - Downloading egulias/email-validator (2.1.25)
7:01:43 AM ***************:   - Downloading swiftmailer/swiftmailer (v6.3.0)
7:01:43 AM ***************:   - Downloading symfony/polyfill-php81 (v1.25.0)
7:01:43 AM ***************:   - Downloading ramsey/collection (1.2.2)
7:01:43 AM ***************:   - Downloading brick/math (0.9.3)
7:01:43 AM ***************:   - Downloading ramsey/uuid (4.3.1)
7:01:43 AM ***************:   - Downloading psr/simple-cache (1.0.1)
7:01:43 AM ***************:   - Downloading opis/closure (3.6.3)
7:01:43 AM ***************:   - Downloading symfony/translation-contracts (v3.0.1)
7:01:43 AM ***************:   - Downloading symfony/translation (v6.0.7)
7:01:43 AM ***************:   - Downloading nesbot/carbon (2.57.0)
7:01:43 AM ***************:   - Downloading league/mime-type-detection (1.11.0)
7:01:43 AM ***************:   - Downloading league/flysystem (1.1.9)
7:01:43 AM ***************:   - Downloading nette/utils (v3.2.7)
7:01:43 AM ***************:   - Downloading nette/schema (v1.2.2)
7:01:43 AM ***************:   - Downloading dflydev/dot-access-data (v3.0.1)
7:01:43 AM ***************:   - Downloading league/config (v1.1.1)
7:01:43 AM ***************:   - Downloading league/commonmark (2.3.0)
7:01:43 AM ***************:   - Downloading laravel/serializable-closure (v1.1.1)
7:01:43 AM ***************:   - Downloading laravel/framework (v8.83.9)
7:01:43 AM ***************:   - Downloading facade/ignition-contracts (1.0.2)
7:01:43 AM ***************:   - Downloading facade/flare-client-php (1.9.1)
7:01:43 AM ***************:   - Downloading facade/ignition (2.17.5)
7:01:43 AM ***************:   - Downloading fakerphp/faker (v1.19.0)
7:01:43 AM ***************:   - Downloading asm89/stack-cors (v2.1.1)
7:01:43 AM ***************:   - Downloading fruitcake/laravel-cors (v2.2.0)
7:01:43 AM ***************:   - Downloading psr/http-message (1.0.1)
7:01:43 AM ***************:   - Downloading psr/http-client (1.0.1)
7:01:43 AM ***************:   - Downloading ralouphie/getallheaders (3.0.3)
7:01:43 AM ***************:   - Downloading psr/http-factory (1.0.1)
7:01:43 AM ***************:   - Downloading guzzlehttp/psr7 (2.2.1)
7:01:43 AM ***************:   - Downloading guzzlehttp/promises (1.5.1)
7:01:43 AM ***************:   - Downloading guzzlehttp/guzzle (7.4.2)
7:01:43 AM ***************:   - Downloading laravel/sail (v1.13.10)
7:01:43 AM ***************:   - Downloading laravel/sanctum (v2.15.1)
7:01:43 AM ***************:   - Downloading nikic/php-parser (v4.13.2)
7:01:43 AM ***************:   - Downloading psy/psysh (v0.11.2)
7:01:43 AM ***************:   - Downloading laravel/tinker (v2.7.2)
7:01:43 AM ***************:   - Downloading hamcrest/hamcrest-php (v2.0.1)
7:01:43 AM ***************:   - Downloading mockery/mockery (1.5.0)
7:01:43 AM ***************:   - Downloading filp/whoops (2.14.5)
7:01:44 AM ***************:   - Downloading nunomaduro/collision (v5.11.0)
7:01:44 AM ***************:   - Downloading phpdocumentor/reflection-common (2.2.0)
7:01:44 AM ***************:   - Downloading phpdocumentor/type-resolver (1.6.1)
7:01:44 AM ***************:   - Downloading phpdocumentor/reflection-docblock (5.3.0)
7:01:44 AM ***************:   - Downloading sebastian/version (3.0.2)
7:01:44 AM ***************:   - Downloading sebastian/type (3.0.0)
7:01:44 AM ***************:   - Downloading sebastian/resource-operations (3.0.3)
7:01:44 AM ***************:   - Downloading sebastian/recursion-context (4.0.4)
7:01:44 AM ***************:   - Downloading sebastian/object-reflector (2.0.4)
7:01:44 AM ***************:   - Downloading sebastian/object-enumerator (4.0.4)
7:01:44 AM ***************:   - Downloading sebastian/global-state (5.0.5)
7:01:44 AM ***************:   - Downloading sebastian/exporter (4.0.4)
7:01:44 AM ***************:   - Downloading sebastian/environment (5.1.4)
7:01:44 AM ***************:   - Downloading sebastian/diff (4.0.4)
7:01:44 AM ***************:   - Downloading sebastian/comparator (4.0.6)
7:01:44 AM ***************:   - Downloading sebastian/code-unit (1.0.8)
7:01:44 AM ***************:   - Downloading sebastian/cli-parser (1.0.1)
7:01:44 AM ***************:   - Downloading phpunit/php-timer (5.0.3)
7:01:44 AM ***************:   - Downloading phpunit/php-text-template (2.0.4)
7:01:44 AM ***************:   - Downloading phpunit/php-invoker (3.1.1)
7:01:44 AM ***************:   - Downloading phpunit/php-file-iterator (3.0.6)
7:01:44 AM ***************:   - Downloading theseer/tokenizer (1.2.1)
7:01:44 AM ***************:   - Downloading sebastian/lines-of-code (1.0.3)
7:01:44 AM ***************:   - Downloading sebastian/complexity (2.0.2)
7:01:44 AM ***************:   - Downloading sebastian/code-unit-reverse-lookup (2.0.3)
7:01:44 AM ***************:   - Downloading phpunit/php-code-coverage (9.2.15)
7:01:44 AM ***************:   - Downloading doctrine/instantiator (1.4.1)
7:01:44 AM ***************:   - Downloading phpspec/prophecy (v1.15.0)
7:01:44 AM ***************:   - Downloading phar-io/version (3.2.1)
7:01:44 AM ***************:   - Downloading phar-io/manifest (2.0.3)
7:01:44 AM ***************:   - Downloading myclabs/deep-copy (1.11.0)
7:01:44 AM ***************:   - Downloading phpunit/phpunit (9.5.20)
7:01:44 AM ***************:    0/110 [>---------------------------]   0%
7:01:44 AM ***************:    1/110 [>---------------------------]   0%
7:01:44 AM ***************:    5/110 [=>--------------------------]   4%
7:01:44 AM ***************:    8/110 [==>-------------------------]   7%
7:01:44 AM ***************:    9/110 [==>-------------------------]   8%
7:01:44 AM ***************:   10/110 [==>-------------------------]   9%
7:01:44 AM ***************:   11/110 [==>-------------------------]  10%
7:01:44 AM ***************:   14/110 [===>------------------------]  12%
7:01:44 AM ***************:   15/110 [===>------------------------]  13%
7:01:44 AM ***************:   16/110 [====>-----------------------]  14%
7:01:44 AM ***************:   17/110 [====>-----------------------]  15%
7:01:44 AM ***************:   19/110 [====>-----------------------]  17%
7:01:44 AM ***************:   20/110 [=====>----------------------]  18%
7:01:44 AM ***************:   21/110 [=====>----------------------]  19%
7:01:45 AM ***************:   23/110 [=====>----------------------]  20%
7:01:45 AM ***************:   25/110 [======>---------------------]  22%
7:01:45 AM ***************:   26/110 [======>---------------------]  23%
7:01:45 AM ***************:   27/110 [======>---------------------]  24%
7:01:45 AM ***************:   28/110 [=======>--------------------]  25%
7:01:45 AM ***************:   31/110 [=======>--------------------]  28%
7:01:45 AM ***************:   33/110 [========>-------------------]  30%
7:01:45 AM ***************:   34/110 [========>-------------------]  30%
7:01:45 AM ***************:   36/110 [=========>------------------]  32%
7:01:45 AM ***************:   37/110 [=========>------------------]  33%
7:01:45 AM ***************:   40/110 [==========>-----------------]  36%
7:01:45 AM ***************:   41/110 [==========>-----------------]  37%
7:01:45 AM ***************:   43/110 [==========>-----------------]  39%
7:01:45 AM ***************:   44/110 [===========>----------------]  40%
7:01:45 AM ***************:   46/110 [===========>----------------]  41%
7:01:45 AM ***************:   47/110 [===========>----------------]  42%
7:01:45 AM ***************:   48/110 [============>---------------]  43%
7:01:45 AM ***************:   50/110 [============>---------------]  45%
7:01:46 AM ***************:   51/110 [============>---------------]  46%
7:01:46 AM ***************:   52/110 [=============>--------------]  47%
7:01:46 AM ***************:   53/110 [=============>--------------]  48%
7:01:46 AM ***************:   54/110 [=============>--------------]  49%
7:01:46 AM ***************:   55/110 [==============>-------------]  50%
7:01:47 AM ***************:   58/110 [==============>-------------]  52%
7:01:47 AM ***************:   60/110 [===============>------------]  54%
7:01:47 AM ***************:   62/110 [===============>------------]  56%
7:01:47 AM ***************:   64/110 [================>-----------]  58%
7:01:47 AM ***************:   67/110 [=================>----------]  60%
7:01:47 AM ***************:   68/110 [=================>----------]  61%
7:01:47 AM ***************:   69/110 [=================>----------]  62%
7:01:47 AM ***************:   70/110 [=================>----------]  63%
7:01:47 AM ***************:   72/110 [==================>---------]  65%
7:01:48 AM ***************:   73/110 [==================>---------]  66%
7:01:48 AM ***************:   74/110 [==================>---------]  67%
7:01:48 AM ***************:   76/110 [===================>--------]  69%
7:01:48 AM ***************:   77/110 [===================>--------]  70%
7:01:48 AM ***************:   79/110 [====================>-------]  71%
7:01:48 AM ***************:   81/110 [====================>-------]  73%
7:01:48 AM ***************:   83/110 [=====================>------]  75%
7:01:48 AM ***************:   84/110 [=====================>------]  76%
7:01:48 AM ***************:   85/110 [=====================>------]  77%
7:01:48 AM ***************:   87/110 [======================>-----]  79%
7:01:48 AM ***************:   88/110 [======================>-----]  80%
7:01:48 AM ***************:   92/110 [=======================>----]  83%
7:01:48 AM ***************:   95/110 [========================>---]  86%
7:01:48 AM ***************:   97/110 [========================>---]  88%
7:01:49 AM ***************:   98/110 [========================>---]  89%
7:01:49 AM ***************:   99/110 [=========================>--]  90%
7:01:49 AM ***************:  102/110 [=========================>--]  92%
7:01:49 AM ***************:  103/110 [==========================>-]  93%
7:01:49 AM ***************:  106/110 [==========================>-]  96%
7:01:49 AM ***************:  108/110 [===========================>]  98%
7:01:49 AM ***************:  109/110 [===========================>]  99%
7:01:49 AM ***************:  110/110 [============================] 100%  - Installing doctrine/inflector (2.0.4): Extracting archive
7:01:49 AM ***************:   - Installing doctrine/lexer (1.2.3): Extracting archive
7:01:49 AM ***************:   - Installing symfony/polyfill-ctype (v1.25.0): Extracting archive
7:01:49 AM ***************:   - Installing webmozart/assert (1.10.0): Extracting archive
7:01:49 AM ***************:   - Installing dragonmantank/cron-expression (v3.3.1): Extracting archive
7:01:49 AM ***************:   - Installing symfony/polyfill-php80 (v1.25.0): Extracting archive
7:01:49 AM ***************:   - Installing symfony/polyfill-mbstring (v1.25.0): Extracting archive
7:01:49 AM ***************:   - Installing symfony/var-dumper (v5.4.6): Extracting archive
7:01:49 AM ***************:   - Installing symfony/polyfill-intl-normalizer (v1.25.0): Extracting archive
7:01:49 AM ***************:   - Installing symfony/polyfill-intl-grapheme (v1.25.0): Extracting archive
7:01:49 AM ***************:   - Installing symfony/string (v6.0.3): Extracting archive
7:01:49 AM ***************:   - Installing symfony/deprecation-contracts (v3.0.1): Extracting archive
7:01:49 AM ***************:   - Installing psr/container (1.1.2): Extracting archive
7:01:49 AM ***************:   - Installing symfony/service-contracts (v2.5.1): Extracting archive
7:01:49 AM ***************:   - Installing symfony/polyfill-php73 (v1.25.0): Extracting archive
7:01:49 AM ***************:   - Installing symfony/console (v5.4.7): Extracting archive
7:01:49 AM ***************:   - Installing psr/log (2.0.0): Extracting archive
7:01:50 AM ***************:   - Installing monolog/monolog (2.5.0): Extracting archive
7:01:50 AM ***************:   - Installing voku/portable-ascii (1.6.1): Extracting archive
7:01:50 AM ***************:   - Installing phpoption/phpoption (1.8.1): Extracting archive
7:01:50 AM ***************:   - Installing graham-campbell/result-type (v1.0.4): Extracting archive
7:01:50 AM ***************:   - Installing vlucas/phpdotenv (v5.4.1): Extracting archive
7:01:50 AM ***************:   - Installing symfony/css-selector (v6.0.3): Extracting archive
7:01:50 AM ***************:   - Installing tijsverkoyen/css-to-inline-styles (2.2.4): Extracting archive
7:01:50 AM ***************:   - Installing symfony/routing (v5.4.3): Extracting archive
7:01:50 AM ***************:   - Installing symfony/process (v5.4.7): Extracting archive
7:01:50 AM ***************:   - Installing symfony/polyfill-php72 (v1.25.0): Extracting archive
7:01:50 AM ***************:   - Installing symfony/polyfill-intl-idn (v1.25.0): Extracting archive
7:01:50 AM ***************:   - Installing symfony/mime (v5.4.7): Extracting archive
7:01:50 AM ***************:   - Installing symfony/http-foundation (v5.4.6): Extracting archive
7:01:50 AM ***************:   - Installing psr/event-dispatcher (1.0.0): Extracting archive
7:01:50 AM ***************:   - Installing symfony/event-dispatcher-contracts (v3.0.1): Extracting archive
7:01:50 AM ***************:   - Installing symfony/event-dispatcher (v6.0.3): Extracting archive
7:01:50 AM ***************:   - Installing symfony/error-handler (v5.4.7): Extracting archive
7:01:50 AM ***************:   - Installing symfony/http-kernel (v5.4.7): Extracting archive
7:01:50 AM ***************:   - Installing symfony/finder (v5.4.3): Extracting archive
7:01:50 AM ***************:   - Installing symfony/polyfill-iconv (v1.25.0): Extracting archive
7:01:50 AM ***************:   - Installing egulias/email-validator (2.1.25): Extracting archive
7:01:50 AM ***************:   - Installing swiftmailer/swiftmailer (v6.3.0): Extracting archive
7:01:50 AM ***************:   - Installing symfony/polyfill-php81 (v1.25.0): Extracting archive
7:01:50 AM ***************:   - Installing ramsey/collection (1.2.2): Extracting archive
7:01:50 AM ***************:   - Installing brick/math (0.9.3): Extracting archive
7:01:50 AM ***************:   - Installing ramsey/uuid (4.3.1): Extracting archive
7:01:50 AM ***************:   - Installing psr/simple-cache (1.0.1): Extracting archive
7:01:50 AM ***************:   - Installing opis/closure (3.6.3): Extracting archive
7:01:50 AM ***************:   - Installing symfony/translation-contracts (v3.0.1): Extracting archive
7:01:50 AM ***************:   - Installing symfony/translation (v6.0.7): Extracting archive
7:01:50 AM ***************:   - Installing nesbot/carbon (2.57.0): Extracting archive
7:01:50 AM ***************:   - Installing league/mime-type-detection (1.11.0): Extracting archive
7:01:50 AM ***************:   - Installing league/flysystem (1.1.9): Extracting archive
7:01:50 AM ***************:   - Installing nette/utils (v3.2.7): Extracting archive
7:01:50 AM ***************:   - Installing nette/schema (v1.2.2): Extracting archive
7:01:50 AM ***************:   - Installing dflydev/dot-access-data (v3.0.1): Extracting archive
7:01:50 AM ***************:   - Installing league/config (v1.1.1): Extracting archive
7:01:50 AM ***************:   - Installing league/commonmark (2.3.0): Extracting archive
7:01:50 AM ***************:   - Installing laravel/serializable-closure (v1.1.1): Extracting archive
7:01:50 AM ***************:   - Installing laravel/framework (v8.83.9): Extracting archive
7:01:50 AM ***************:   - Installing facade/ignition-contracts (1.0.2): Extracting archive
7:01:50 AM ***************:   - Installing facade/flare-client-php (1.9.1): Extracting archive
7:01:50 AM ***************:   - Installing facade/ignition (2.17.5): Extracting archive
7:01:50 AM ***************:   - Installing fakerphp/faker (v1.19.0): Extracting archive
7:01:50 AM ***************:   - Installing asm89/stack-cors (v2.1.1): Extracting archive
7:01:51 AM ***************:   - Installing fruitcake/laravel-cors (v2.2.0): Extracting archive
7:01:51 AM ***************:   - Installing psr/http-message (1.0.1): Extracting archive
7:01:51 AM ***************:   - Installing psr/http-client (1.0.1): Extracting archive
7:01:51 AM ***************:   - Installing ralouphie/getallheaders (3.0.3): Extracting archive
7:01:51 AM ***************:   - Installing psr/http-factory (1.0.1): Extracting archive
7:01:51 AM ***************:   - Installing guzzlehttp/psr7 (2.2.1): Extracting archive
7:01:51 AM ***************:   - Installing guzzlehttp/promises (1.5.1): Extracting archive
7:01:51 AM ***************:   - Installing guzzlehttp/guzzle (7.4.2): Extracting archive
7:01:51 AM ***************:   - Installing laravel/sail (v1.13.10): Extracting archive
7:01:51 AM ***************:   - Installing laravel/sanctum (v2.15.1): Extracting archive
7:01:51 AM ***************:   - Installing nikic/php-parser (v4.13.2): Extracting archive
7:01:51 AM ***************:   - Installing psy/psysh (v0.11.2): Extracting archive
7:01:51 AM ***************:   - Installing laravel/tinker (v2.7.2): Extracting archive
7:01:51 AM ***************:   - Installing hamcrest/hamcrest-php (v2.0.1): Extracting archive
7:01:51 AM ***************:   - Installing mockery/mockery (1.5.0): Extracting archive
7:01:51 AM ***************:   - Installing filp/whoops (2.14.5): Extracting archive
7:01:51 AM ***************:   - Installing nunomaduro/collision (v5.11.0): Extracting archive
7:01:51 AM ***************:   - Installing phpdocumentor/reflection-common (2.2.0): Extracting archive
7:01:51 AM ***************:   - Installing phpdocumentor/type-resolver (1.6.1): Extracting archive
7:01:51 AM ***************:   - Installing phpdocumentor/reflection-docblock (5.3.0): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/version (3.0.2): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/type (3.0.0): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/resource-operations (3.0.3): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/recursion-context (4.0.4): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/object-reflector (2.0.4): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/object-enumerator (4.0.4): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/global-state (5.0.5): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/exporter (4.0.4): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/environment (5.1.4): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/diff (4.0.4): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/comparator (4.0.6): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/code-unit (1.0.8): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/cli-parser (1.0.1): Extracting archive
7:01:51 AM ***************:   - Installing phpunit/php-timer (5.0.3): Extracting archive
7:01:51 AM ***************:   - Installing phpunit/php-text-template (2.0.4): Extracting archive
7:01:51 AM ***************:   - Installing phpunit/php-invoker (3.1.1): Extracting archive
7:01:51 AM ***************:   - Installing phpunit/php-file-iterator (3.0.6): Extracting archive
7:01:51 AM ***************:   - Installing theseer/tokenizer (1.2.1): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/lines-of-code (1.0.3): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/complexity (2.0.2): Extracting archive
7:01:51 AM ***************:   - Installing sebastian/code-unit-reverse-lookup (2.0.3): Extracting archive
7:01:51 AM ***************:   - Installing phpunit/php-code-coverage (9.2.15): Extracting archive
7:01:51 AM ***************:   - Installing doctrine/instantiator (1.4.1): Extracting archive
7:01:51 AM ***************:   - Installing phpspec/prophecy (v1.15.0): Extracting archive
7:01:51 AM ***************:   - Installing phar-io/version (3.2.1): Extracting archive
7:01:51 AM ***************:   - Installing phar-io/manifest (2.0.3): Extracting archive
7:01:52 AM ***************:   - Installing myclabs/deep-copy (1.11.0): Extracting archive
7:01:52 AM ***************:   - Installing phpunit/phpunit (9.5.20): Extracting archive
7:01:52 AM ***************:    0/100 [>---------------------------]   0%
7:01:52 AM ***************:   10/100 [==>-------------------------]  10%
7:01:52 AM ***************:   20/100 [=====>----------------------]  20%
7:01:52 AM ***************:   30/100 [========>-------------------]  30%
7:01:53 AM ***************:   36/100 [==========>-----------------]  36%
7:01:54 AM ***************:   46/100 [============>---------------]  46%
7:01:55 AM ***************:   52/100 [==============>-------------]  52%
7:01:55 AM ***************:   61/100 [=================>----------]  61%
7:01:56 AM ***************:   69/100 [===================>--------]  69%
7:01:56 AM ***************:   79/100 [======================>-----]  79%
7:02:00 AM ***************:   89/100 [========================>---]  89%
7:02:00 AM ***************:   99/100 [===========================>]  99%
7:02:00 AM ***************:  100/100 [============================] 100%Package swiftmailer/swiftmailer is abandoned, you should avoid using it. Use symfony/mailer instead.
7:02:00 AM ***************: Generating optimized autoload files
7:02:03 AM ***************: > Illuminate\Foundation\ComposerScripts::postAutoloadDump
7:02:03 AM ***************: > @php artisan package:discover --ansi
7:02:04 AM ***************: Discovered Package: [32mfacade/ignition[39m
7:02:04 AM ***************: Discovered Package: [32mfruitcake/laravel-cors[39m
7:02:04 AM ***************: 77 packages you are using are looking for funding.
7:02:04 AM ***************: Discovered Package: [32mlaravel/sail[39m
7:02:04 AM ***************: Use the `composer fund` command to find out more!
7:02:04 AM ***************: Discovered Package: [32mlaravel/sanctum[39m
7:02:04 AM ***************: Discovered Package: [32mlaravel/tinker[39m
7:02:04 AM ***************: Discovered Package: [32mnesbot/carbon[39m
7:02:04 AM ***************: Discovered Package: [32mnunomaduro/collision[39m
7:02:04 AM ***************: [32mPackage manifest generated successfully.[39m
7:02:04 AM ***************: Preparing output...
7:02:04 AM ***************: Copying files to destination directory '/home/site/wwwroot'...
7:03:55 AM ***************: Done in 111 sec(s).
7:03:55 AM ***************: Removing existing manifest file
7:03:55 AM ***************: Creating a manifest file...
7:03:55 AM ***************: Manifest file created.
7:03:55 AM ***************: Done in 135 sec(s).
7:03:57 AM ***************: Running post deployment command(s)...
7:03:58 AM ***************: Triggering recycle (preview mode disabled).
7:03:58 AM ***************: Deployment successful.
```
</details>

![image](https://user-images.githubusercontent.com/4566555/164838224-f3f8c9cc-4b43-4ab3-ab49-258886bbc896.png)

Reply:
Hi @horihiro , thanks for providing information. Not sure which image you're using in this build. If it's with VSCode, it could be either the vso image or vso-focal image. App service uses a different build image. I tried the prebuild script in github-action buster image but doesn't seem to work there.

Reply:
Update: This workaround worked when I test locally, thanks for your idea! So we'll need to install debian image version specific SDK from https://oryx-cdn.microsoft.io/php/php-**{debian version}**-8.0.x.tar.gz. We'll need to find ways to identify the image version type, then install the corresponding SDK.